src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -1443,7 +1443,12 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
   }
 
   antlrcpp::Any visitIri(SparqlAutomaticParser::IriContext* ctx) override {
-    string langtag = ctx->LANGTAG() ? ctx->LANGTAG()->getText() + '@' : "";
+    return visitTypesafe(ctx);
+  }
+
+  string visitTypesafe(SparqlAutomaticParser::IriContext* ctx) {
```

Comment:
Why is this function called `visitTypesafe` and why is this code in a separate function and not in `visitIri`?

I realize that `visitTypesafe` has already been used a lot (with different signatures) in the previous version of the code. So this is not so much a question about this PR, but about that concept in general.

Reply:
## TL;DR:
It is an artifact of making the Visitor typesafe. It is still in progress so there are both typesafe (`visitTypesafe`) and the regular (`visit` and `visit...`) functions at the moment. When this migration is complete the current `visit` and `visit...` functions can be removed and `visitTypesafe` be renamed to `visit`. But currently `visit...` is still needed so we just call the typesafe function from it.

## Long explanaition:
ANTLR wraps return values in `antlrcpp::Any`. It does this because it provides a function `visit` which parses any Clause you give it. The return value is wrapped in the `Any` type. We would prefer to do this in a typesafe way. The Idea is to replace the `visit` that accepts any Context and could return any type with overloaded functions that process a specific clause and have a specific return type.
So instead of
```
antlrcpp::Any visit(ParseTree *tree);
```
in the end we would have
```
std::optional<GraphPatternOperation::Values> visit(SparqlAutomaticParser::ValuesClauseContext* ctx);
GraphPatternOperation::Values visit(SparqlAutomaticParser::DataBlockContext* ctx);
SparqlValues visit(SparqlAutomaticParser::InlineDataOneVarContext* ctx);
...
```
This way `visit` can still be called with any context but we known the return type. If all functions are using this the `Typesafe` suffix can be removed. This would also enable us to remove the depency to `SparqlAutomaticVisitor` in our visitor.

Reply:
In addition:
an advantage of having a generic `visit` function (no matter if it's called visit or visitTypesafe) is that it makes generic code
that calls and wraps the parser much simpler vs the somewhat redundant `visitIriOrFunction(iriOrFunctionContext*)` which is the default of ANTLR.